src/System.Linq.Expressions/src/System/Linq/Expressions/ConstantExpression.cs

```diff
@@ -14,29 +14,16 @@ namespace System.Linq.Expressions
     [DebuggerTypeProxy(typeof(ConstantExpressionProxy))]
     public class ConstantExpression : Expression
     {
-        private readonly object _value;
-
         internal ConstantExpression(object value)
         {
-            _value = value;
+            Value = value;
         }
 
         /// <summary>
         /// Gets the static type of the expression that this <see cref="Expression"/> represents.
         /// </summary>
         /// <returns>The <see cref="System.Type"/> that represents the static type of the expression.</returns>
-        public override Type Type
-        {
-            get
-            {
-                if (_value == null)
-                {
-                    return typeof(object);
-                }
-
-                return _value.GetType();
-            }
-        }
+        public override Type Type => Value?.GetType() ?? typeof(object);
```

Comment:
This isn't exactly the same, in that it adds an extra branch (the generated code needs to check whether Value is null but then also whether the result of GetType() is null).  Note that the change from _value to Value also introduces a property evaluation, though it should be inlined to be the same field access by the JIT.

Before:
```
  // Code size       31 (0x1f)
  .maxstack  8
  IL_0000:  ldarg.0
  IL_0001:  ldfld      object Program::_value
  IL_0006:  brtrue.s   IL_0013
  IL_0008:  ldtoken    [mscorlib]System.Object
  IL_000d:  call       class [mscorlib]System.Type [mscorlib]System.Type::GetTypeFromHandle(valuetype [mscorlib]System.RuntimeTypeHandle)
  IL_0012:  ret
  IL_0013:  ldarg.0
  IL_0014:  ldfld      object Program::_value
  IL_0019:  callvirt   instance class [mscorlib]System.Type [mscorlib]System.Object::GetType()
  IL_001e:  ret
```
After:
```
  // Code size       33 (0x21)
  .maxstack  8
  IL_0000:  ldarg.0
  IL_0001:  call       instance object Program::get_Value()
  IL_0006:  dup
  IL_0007:  brtrue.s   IL_000d
  IL_0009:  pop
  IL_000a:  ldnull
  IL_000b:  br.s       IL_0012
  IL_000d:  call       instance class [mscorlib]System.Type [mscorlib]System.Object::GetType()
  IL_0012:  dup
  IL_0013:  brtrue.s   IL_0020
  IL_0015:  pop
  IL_0016:  ldtoken    [mscorlib]System.Object
  IL_001b:  call       class [mscorlib]System.Type [mscorlib]System.Type::GetTypeFromHandle(valuetype [mscorlib]System.RuntimeTypeHandle)
  IL_0020:  ret
```